pkg/event/events.go

```diff
@@ -7,29 +7,39 @@ import (
 	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
 	kyvernov2alpha1 "github.com/kyverno/kyverno/api/kyverno/v2alpha1"
 	engineapi "github.com/kyverno/kyverno/pkg/engine/api"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 func NewPolicyFailEvent(source Source, reason Reason, engineResponse engineapi.EngineResponse, ruleResp engineapi.RuleResponse, blocked bool) Info {
 	action := ResourcePassed
 	if blocked {
 		action = ResourceBlocked
 	}
-
 	pol := engineResponse.Policy()
-
+	regarding := corev1.ObjectReference{
+		// TODO: iirc it's not safe to assume api version is set
+		APIVersion: "kyverno.io/v1",
```

Comment:
This needs to be updated when migrate to v2.

Reply:
Not sure if we need the correct version TBH but yeah ideally we would use the stored version for that.